### PR TITLE
Add FXIOS-8222, DISCO-2718 [v124] add score to sponsored suggestions

### PR DESCRIPTION
**Pull request by tiftran**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-8222)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/17408)

## :bulb: Description
We added score to AMP suggestions in https://github.com/mozilla/application-services/pull/6055 so making changes here to accommodate it, I believe the application services PR needs to be merged before this one, so putting this as draft for now

## :pencil: Checklist
You have to check all boxes before merging
- [ ] Filled in the above information (tickets numbers and description of your work)
- [ ] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

**Comment by tiftran** [comment]
tagging @linabutler , @adudenamedruby , @OrlaM  for a review for another fx suggest related PR

File: firefox-ios/Storage/Rust/RustFirefoxSuggestion.swift
Comment (linabutler) [inline firefox-ios/Storage/Rust/RustFirefoxSuggestion.swift:24]: Hmmm, was there a reason you added this to `RustFirefoxSuggestion`? It doesn't look like we're using the score anywhere else.

File: firefox-ios/Storage/Rust/RustFirefoxSuggestion.swift
Comment (linabutler) [inline firefox-ios/Storage/Rust/RustFirefoxSuggestion.swift:56]: Great!

**Review by linabutler — APPROVED** [review]
r+wc!

File: firefox-ios/Storage/Rust/RustFirefoxSuggestion.swift
Comment (tiftran) [reply firefox-ios/Storage/Rust/RustFirefoxSuggestion.swift:24]: oh good point

**Review by adudenamedruby — APPROVED** [review]

**Comment by adudenamedruby** [comment]
Screenshot 2024-01-23 at 10 02 46 AM

**Comment by tiftran** [comment]
@adudenamedruby silly question but  I think the errors in your screenshot was expected since, the application services PR hadn't landed yet?

**Comment by adudenamedruby** [comment]
Does that mean that this PR is blocked until it lands?

**Comment by adudenamedruby** [comment]
LOL, I just noticed that this PR is still in draft

nevermind!

**Comment by lmarceau** [comment]
Cherry picked those commits onto https://github.com/mozilla-mobile/firefox-ios/pull/18383

**Pull request closed without merging**
